ob-jq.el

```diff
@@ -49,52 +50,85 @@
     (:cmd-line . :any))
   "Jq specific header arguments.")
 
-(defvar org-babel-default-header-args:jq '()
+(defvar org-babel-default-header-args:jq '(
+                                           (:results . "output")
+                                           )
   "Default arguments for evaluating a jq source block.")
 
+(defun org-babel-jq-table-to-json (data)
+  "Convert org table to JSON.
+
+First line specifies the keys."
+  (let* ((header (car data))
+         (data (cdr data)))
+    (while (eq (car data) 'hline)
+      (setq data (cdr data)))
+    (json-encode
+     (mapcar
+      (lambda (row) (cl-mapcar 'cons header row))
+      data))))
+
 (defun org-babel-execute:jq (body params)
   "Execute a block of jq code with org-babel.  This function is
 called by `org-babel-execute-src-block'"
   (message "executing jq source code block")
   (let* ((result-params (cdr (assq :result-params params)))
-	 (cmd-line (cdr (assq :cmd-line params)))
-	 (in-file (cdr (assq :in-file params)))
-	 (code-file (let ((file (org-babel-temp-file "jq-")))
-		      (with-temp-file file
-			(insert body)
-			file)))
-	 (stdin (let ((stdin (cdr (assq :stdin params))))
-		  (when stdin
-		    (let ((tmp (org-babel-temp-file "jq-stdin-"))
-			  (res (org-babel-ref-resolve stdin)))
-		      (with-temp-file tmp
-			(insert res)
-			tmp)))))
-	 (cmd (mapconcat #'identity
-			 (remq nil
-			       (list org-babel-jq-command
-				     (format "--from-file \"%s\"" code-file)
-				     cmd-line
-				     in-file))
-			 " ")))
+         (cmd-line (cdr (assq :cmd-line params)))
+         (in-file (cdr (assq :in-file params)))
+         (code-file (let ((file (org-babel-temp-file "jq-")))
+                      (with-temp-file file
+                        (insert body)
+                        file)))
+         (stdin (let ((stdin (cdr (assq :stdin params))))
+                  (when stdin
+                    (let ((tmp (org-babel-temp-file "jq-stdin-"))
+                          (res (org-babel-ref-resolve stdin)))
+                      (with-temp-file tmp
+                        (insert
+                         (cond
+                          ((stringp res) res)
+                          ((listp res) (org-babel-jq-table-to-json res))))
```

Comment:
I couldn't figure it out from the `org-babe-ref-resolve`; do we know that these two cases in the `cond` are the only cases that can happen? Could it return a number or other datatype? I don't mind not guarding for all of them, but it would help in debugging to have a better error there if someone comes asking. Right now if that happens it will give a very unfriendly error.

Reply:
Agreed, I think it could return a number.  I think we can handle just a list or anything else which will just get passed along for org to do whatever it wants.  Good catch.

Reply:
Yes. When I look at the previous source that is what we do. Just add (t res) as a clause in the cond and we should be good.